src/main/scala/Executor.scala

```diff
@@ -24,5 +24,5 @@ trait Executor {
 }
 
 case class Result(
-  finalValue: AnyVal
+  finalValue: Option[List[AnyVal]]
```

Comment:
AnyVal? Don't like this. What are the possible return types? Int and Double? Maybe we have should have a Result object that gets returned, listing the value and the type.